### ISSUE TITLE
`redundant_pub_crate` false positive on local macros

### ISSUE THREAD

**Issue by digama0** [rust-lang/rust-clippy#8756]
### Summary

This is similar to #8732, but it triggers even without a re-export. Here we are defining a module-scoped macro `foo` inside private module `test`, and the error message says that it is a `pub(crate)` macro (which is false, you can't access the macro from another file even if `test` was pub because it is not `#[macro_export]`), and you should use `pub` instead (which does not match the intended meaning of the code, it's a private macro) (and this is also not syntactically correct, `pub macro_rules!` is not legal).

### Lint Name

redundant_pub_crate

### Reproducer

I tried this code:

```rust
#![warn(clippy::redundant_pub_crate)]
mod test {
  macro_rules! foo { () => {} }
  foo! {}
}
fn main() {}
```

I saw this happen:

```
warning: pub(crate) macro inside private module
 --> src/main.rs:3:3
  |
3 |   macro_rules! foo { () => {} }
  |   -^^^^^^^^^^^^^^^
  |   |
  |   help: consider using: `pub`
  |
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![warn(clippy::redundant_pub_crate)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pub_crate
```

### Version

```text
rustc 1.62.0-nightly (69a5d2481 2022-04-27)
binary: rustc
commit-hash: 69a5d2481e856a5a18885390b8cf6950b9ff8dd3
commit-date: 2022-04-27
host: x86_64-unknown-linux-gnu
release: 1.62.0-nightly
LLVM version: 14.0.1
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by Serial-ATA** [comment]
Pretty sure https://github.com/rust-lang/rust-clippy/pull/8736 fixed this, I can't reproduce it.

**Comment by digama0** [comment]
Just pulled the latest rustc, the issue still seems to happen:
```
$ rustc +nightly -Vv
rustc 1.62.0-nightly (a707f4010 2022-04-29)
binary: rustc
commit-hash: a707f401074bc769bab4efb2bfdde7f6c5a4068d
commit-date: 2022-04-29
host: x86_64-unknown-linux-gnu
release: 1.62.0-nightly
LLVM version: 14.0.1

$ cargo +nightly clippy
    Checking rust-test v0.1.0 (/rust-test)
warning: pub(crate) macro inside private module
 --> src/main.rs:3:3
  |
3 |   macro_rules! foo { () => {} }
  |   -^^^^^^^^^^^^^^^
  |   |
  |   help: consider using: `pub`
  |
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![warn(clippy::redundant_pub_crate)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pub_crate

warning: `rust-test` (bin "rust-test") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s

**Comment by y21** [comment]
This issue seems to be fixed? None of the test cases trigger anymore: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a31d64a5ff10926dd80c2c97a03b2978

**Comment by digama0** [comment]
Confirmed, the tests seem to pass after `nightly-2022-05-06`. Looking at [what happened on that day](https://github.com/rust-lang/rust-clippy/commits/master@%7B2022-05-05%7D) I can only guess it was something that changed in the upstream nightly bump 7e017db.
